### ISSUE TITLE
Add support for default environment in Composition

### ISSUE THREAD

**Issue by bobh66** [crossplane/crossplane#4274]
### What problem are you facing?
When using the Composition environment it would be very helpful to have a method for defining a default set of values that the environment should be initialized with before selection of EnvironmentConfigs.  This could be used to simplify patching of constant values throughout the composition, or to provide default values that may or may not be overridden by the selected EnvironmentConfig(s)

### How could Crossplane help solve your problem?
Add support for a "default" or "initial" environment context that would serve as the starting point for all subsequent EnvironmentConfig selection and patching.  The selected EnvironmentConfig(s) should be merged into the default environment, overwriting duplicate keys and leaving existing keys untouched.

A mergeOptions policy is probably necessary to specify the behavior for `appendSlice` and `keepMapValues`

There has also been discussion of adding support for a schema definition for the Composition environment, which would make it easier to understand and validate the expected environment structure, and could also provide a mechanism for specifying default values.  These two features can probably exist in parallel, with the schema validating the initial configuration as well as specifying default values.

**Comment by MisterMX** [comment]
This could be solved easily by adding a new `spec.environment.initData` property to the composition and just include it in the config merge:

```yaml
spec:
  environment:
    initData:
      region: us-west-1
      accountId: 12345665
```

**Comment by MisterMX** [comment]
Having a second thought about this, I got a bit skeptical about this feature. The goal of environments was to separate code and configuration. By adding this, we are practically reversing this change.

Since default configuration can be easily achieved by referencing a default environment config or directly defining default values or hardcoding them int them composition directly.

To me this seems like an anti-pattern and I would therefore vote for not implementing this.

**Comment by MisterMX** [comment]
@bobh66 I guess you are right and it is probably good to add default environment values to the compositions. Maybe I was thinking to dogmatic here ;-)
